ui/raidboss/emulator/data/Encounter.ts

```diff
@@ -1,56 +1,78 @@
 import CombatantTracker from './CombatantTracker';
-import LogEventHandler from './LogEventHandler';
 import PetNamesByLang from '../../../../resources/pet_names';
 import EmulatorCommon from '../EmulatorCommon';
 import LogRepository from './network_log_converter/LogRepository';
 import NetworkLogConverter from './NetworkLogConverter';
+import { Lang } from 'types/global';
+import LineEvent, { isLineEventSource, isLineEventTarget } from './network_log_converter/LineEvent';
+import { UnreachableCode } from '../../../../resources/not_reached';
 
-const isPetName = (name, language = undefined) => {
+const isPetName = (name: string, language: Lang | undefined = undefined) => {
   if (language)
     return PetNamesByLang[language].includes(name);
 
   for (const lang in PetNamesByLang) {
-    if (PetNamesByLang[lang].includes(name))
+    if (PetNamesByLang[lang as Lang].includes(name))
```

Comment:
We could also add a helper function somewhere that converts `string` to `Lang` (and throws otherwise) for cases where you know a string is a lang?

Alternatively, something like this seems better, as it doesn't assume the key type, just converts from string to whatever the key type is:

```typescript
for (const langStr in PetNamesByLang)
  const lang = langStr as keyof typeof PetNamesByLang;
```

Reply:
I think a typeguard would work fine here, but it should probably live in `global.d.ts`?

Reply:
Sounds good!

Reply:
You want that in this PR, or you want a separate PR and then a rebase for this one?

Reply:
I think this PR is fine.  It's really only a couple of lines.